frontend/src/lib/WebsocketConnection.tsx

```diff
@@ -374,10 +378,12 @@ export class WebsocketConnection {
     // Sec-WebSocket-Protocol is set, many clients expect the server to respond
     // with a selected subprotocol to use. We don't want that reply to be the
     // auth token, so we just hard-code it to "streamlit".
-    const hostAuthToken = this.args.getHostAuthToken()
+
+    const sessionToken =
+      this.args.getHostAuthToken() || SessionInfo.lastSessionId
```

Comment:
This is unexpected enough ("lastSessionId" can double as an auth token?) that it feels like it deserves its own `getSessionToken` utility function, with a short docstring explaining why `lastSessionId` is used here.

Reply:
👍 done